p2p/src/network/pubsub/p2p_network_pubsub_reducer.rs

```diff
@@ -123,14 +123,26 @@ impl P2pNetworkPubsubState {
                 peer_id,
                 data,
                 seen_limit,
+                addr,
                 ..
             } => {
                 pubsub_state.reduce_incoming_data(&peer_id, data, meta.time())?;
 
-                let dispatcher: &mut Dispatcher<Action, State> = state_context.into_dispatcher();
+                let (dispatcher, state) = state_context.into_dispatcher_and_state();
+                let p2p_state: &P2pState = state.substate()?;
+                let state = &p2p_state.network.scheduler.broadcast_state;
+                let Some(state) = state.clients.get(&peer_id) else {
+                    // TODO: investigate, cannot reproduce this
+                    // bug_condition!("{:?} not found in state.clients", peer_id);
```

Comment:
should this be silently ignored? do you remember what was the context?

Reply:
This was created by @vlad9486 so I just copied the comment, but since it is bug condition I could leave it in code and it would just log error in production